### PR TITLE
make userName configurable

### PR DESCRIPTION
**Pull request by flarno11**
this PR is needed to support OpenID providers that don't have a "name" claim

### CODE REVIEW

File: connector/oidc/oidc.go [hunk 1]
```diff
@@ -209,9 +214,13 @@ func (c *oidcConnector) HandleCallback(s connector.Scopes, r *http.Request) (ide
 		return identity, fmt.Errorf("oidc: failed to decode claims: %v", err)
 	}
 
-	name, found := claims["name"].(string)
+	var userNameKey = "name"
```
Comment (srenatus) [inline connector/oidc/oidc.go]: just nitpicking: 
```suggestion
	userNameKey := "name"
```

File: connector/oidc/oidc.go [hunk 2]
```diff
@@ -209,9 +214,13 @@ func (c *oidcConnector) HandleCallback(s connector.Scopes, r *http.Request) (ide
...
+	if c.userNameKey != "" {
+		userNameKey = c.userNameKey
+	}
+	name, found := claims[userNameKey].(string)
 	if !found {
-		return identity, errors.New("missing \"name\" claim")
+		return identity, errors.New(fmt.Sprintf("missing \"%s\" claim", userNameKey))
```
Comment (srenatus) [inline connector/oidc/oidc.go]:
```suggestion
		return identity, fmt.Errorf("missing \"%s\" claim", userNameKey)
```

Comment (srenatus) [reply connector/oidc/oidc.go]: hehe, looks like CI agrees: https://travis-ci.org/dexidp/dex/builds/538921410#L1211

**Comment by srenatus** [comment]
@flarno11 thanks for bearing with me. Would you mind squashing these commits? 😃 Happy to mash the button then.

**Comment by flarno11** [comment]
@srenatus I've squashed the commits

**Comment by flarno11** [comment]
@srenatus cool thanks! is there a roadmap for a next release already?

**Comment by srenatus** [comment]
#1458 wouldn't go so far to call it a road map, but this will happen soon, I suppose 😄

**Comment by srenatus** [comment]
I've been meaning to review #1454 but haven't found the time yet...

**Pull request merged**